core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -863,43 +863,71 @@ object SQLPlanParser extends Logging {
+    // SortMergeJoin [name#11, CEIL(dept#12)], [name#28, CEIL(dept_id#27)], Inner
+    // BroadcastHashJoin [name#11, CEIL(dept#12)], [name#28, CEIL(dept_id#27)], Inner,
+    // BuildRight, false
+    // BroadcastHashJoin exprString: [i_item_id#56], [i_item_id#56#86], ExistenceJoin(exists#86),
+    // BuildRight
+    val parsedExpressions = ArrayBuffer[String]()
+    // Get all the join expressions and split it with delimiter :: so that it could be used to
+    // parse function names (if present) later.
+    val joinExprs = equiJoinRegexPattern.findAllMatchIn(exprStr).mkString("::")
+    // Get joinType and buildSide(if applicable)
+    val joinParams = equiJoinRegexPattern.replaceAllIn(
+      exprStr, "").split(",").map(_.trim).filter(_.nonEmpty)
+    val joinType = if (joinParams.nonEmpty) {
+      joinParams(0).split("\\(")(0).trim
+    } else {
+      ""
+    }
+
+    // This is to differentiate between SortMergeJoin and other Joins. SortMergeJoin has no
+    // buildSide.
+    val possibleBuildSides = Set(BuildSide.BuildLeft, BuildSide.BuildRight)
+    val buildSide = joinParams.find(possibleBuildSides.contains).getOrElse("")
+    val isSortMergeJoin = buildSide.isEmpty
+
+    val joinCondition = joinParams.dropWhile(
+      param => possibleBuildSides.contains(param) || param.contains(joinType)).map(_.trim)
+    // Get individual expressions which is later used to get the function names.
+    val colExpressions = joinExprs.split("::").map(_.trim).map(
+      _.replaceAll("""^\[+|\]+$""", "")).map(_.split(",")).flatten.map(_.trim)
+    colExpressions.foreach(expr => addFunctionNames(expr, parsedExpressions))
+    if (joinCondition.nonEmpty) {
+      val conditionExprs = parseConditionalExpressions(joinCondition.mkString(" "))
+      conditionExprs.foreach(parsedExpressions += _)
+    }
+    // Check corner cases for SortMergeJoin
+    val isSortMergeSupported = !(isSortMergeJoin &&
+        joinCondition.nonEmpty && isSMJConditionUnsupported(joinCondition.mkString(" ")))
+
+    (parsedExpressions.distinct.toArray, equiJoinSupportedTypes(buildSide, joinType)
+        && isSortMergeSupported)
+  }
+
+  def isSMJConditionUnsupported(joinCondition: String): Boolean = {
+    // TODO: This is a temporary solution to check for unsupported conditions in SMJ.
+    // Remove these checks once below issues are resolved:
+    // https://github.com/NVIDIA/spark-rapids/issues/11213
+    // https://github.com/NVIDIA/spark-rapids/issues/11214
+
+    // Split the joinCondition by logical operators (AND/OR)
+    val conditions = joinCondition.split("\\s+(AND|OR)\\s+").map(_.trim)
+    conditions.exists { condition =>
+      val normalizedCondition = condition.toLowerCase
+      // Check for the specific corner cases that mark the SMJ as not supported
+      (normalizedCondition.contains("cast") && normalizedCondition.contains("as date")) ||
+          (normalizedCondition.contains("lower") && normalizedCondition.contains("in"))
```

Comment:
Is this possible that "in" is part of the data frame/column name and this could be a false positive? It may be edge case, e.g. `lower(a.link) in ('https://xxxx','https://xyzxyz')`.

Reply:
`normalizedCondition.contains("in")` will match any token with `in` it which will causes too many false positives.
Even if this is replaced by equals `in(/in` , I am skeptical it is a valid condition because the conditions cannot verify the relative positioning of each token which is the key driver for that corner case.

Reply:
We can probably use regex to capture the relative positioning.

Reply:
Thanks! Updated it to use regex for these corner cases:

`((cast(birthday_str#33 as date) >= birthday_dt#47), (cast(birthday_str#33 as date) <= date_add(birthday_dt#47 1)))` -> `true`
`lower(id#30) IN (1,2)` -> `true`
`lower(id#30) IN subquery#70` -> `true`
`lower(id#30)IN(1,2)` -> `false`
`lower(a.link)` -> `false`